pkg/clustermesh/clustermesh.go

```diff
@@ -119,113 +112,30 @@ func IsClusterConfigRequired(ctx context.Context, backend kvstore.BackendOperati
 	return val != nil, err
 }
 
-// RemoteIdentityWatcher is any type which provides identities that have been
-// allocated on a remote cluster.
-type RemoteIdentityWatcher interface {
-	// WatchRemoteIdentities starts watching for identities in another kvstore and
-	// syncs all identities to the local identity cache. RemoteName should be unique
-	// unless replacing an existing remote's backend.
-	WatchRemoteIdentities(remoteName string, backend kvstore.BackendOperations) (*allocator.RemoteCache, error)
-
-	// RemoveRemoteIdentities removes any reference to a remote identity source.
-	RemoveRemoteIdentities(name string)
-
-	// Close stops the watcher.
-	Close()
-}
-
-// ClusterMesh is a cache of multiple remote clusters
-type ClusterMesh struct {
-	// conf is the configuration, it is immutable after NewClusterMesh()
+// GenericClusterMesh is a cache of multiple remote clusters
+type GenericClusterMesh struct {
+	// conf is the configuration, it is immutable after NewGenericClusterMesh()
 	conf Configuration
 
 	mutex         lock.RWMutex
 	clusters      map[string]*remoteCluster
 	configWatcher *configDirectoryWatcher
-
-	// globalServices is a list of all global services. The datastructure
-	// is protected by its own mutex inside the structure.
-	globalServices *globalServiceCache
-
-	// nodeName is the name of the local node. This is used for logging and metrics
-	nodeName string
-
-	// metricTotalRemoteClusters is gauge metric keeping track of total number
-	// of remote clusters.
-	metricTotalRemoteClusters *prometheus.GaugeVec
-
-	// metricLastFailureTimestamp is a gauge metric tracking the last failure timestamp
-	metricLastFailureTimestamp *prometheus.GaugeVec
-
-	// metricReadinessStatus is a gauge metric tracking the readiness status of a remote cluster
-	metricReadinessStatus *prometheus.GaugeVec
-
-	// metricTotalFailure is a gauge metric tracking the number of failures when connecting to a remote cluster
-	metricTotalFailures *prometheus.GaugeVec
-
-	// metricTotalNodes is a gauge metric tracking the number of total nodes in a remote cluster
-	metricTotalNodes *prometheus.GaugeVec
 }
 
-// NewClusterMesh creates a new remote cluster cache based on the
+// NewGenericClusterMesh creates a new remote cluster cache based on the
 // provided configuration
-func NewClusterMesh(lifecycle hive.Lifecycle, c Configuration) *ClusterMesh {
-	if c.ClusterID == 0 {
-		return nil
+func NewGenericClusterMesh(c Configuration) GenericClusterMesh {
+	return GenericClusterMesh{
+		conf:     c,
+		clusters: map[string]*remoteCluster{},
 	}
+}
 
-	if c.ClusterMeshConfig == "" {
+func (cm *GenericClusterMesh) Start(hive.HookContext) error {
+	if cm.conf.ClusterMeshConfig == "" {
```

Comment:
What's the situation that `ClusterMeshConfig == ""`? Can't we move this check to `NewGenericClusterMesh`?